packages/js/product-editor/src/components/attribute-control/utils.ts

```diff
@@ -48,3 +53,36 @@ export function reorderSortableProductAttributePositions(
 		}
 	);
 }
+
+/**
+ * Checks if the given attribute has
+ * either terms (global attributes) or options (local attributes).
+ *
+ * @param {EnhancedProductAttribute} attribute - The attribute to check.
+ * @return {boolean} True if the attribute has terms or options, false otherwise.
+ */
+export const hasTermsOrOptions = (
+	attribute: EnhancedProductAttribute
+): boolean => {
+	return !! (
+		( attribute?.terms && attribute.terms.length > 0 ) ||
+		( attribute?.options && attribute.options.length > 0 )
+	);
+};
+
+/**
+ * Checks if the given attribute is filled out,
+ * meaning it has a name and either terms or options.
+ *
+ * @param {EnhancedProductAttribute | null} attribute - The attribute to check.
+ * @return {attribute is EnhancedProductAttribute} - True if the attribute is filled out, otherwise false.
+ */
+export const isAttributeFilledOut = (
+	attribute: EnhancedProductAttribute | null
+): attribute is EnhancedProductAttribute => {
+	return (
+		attribute !== null &&
+		attribute.name.length > 0 &&
```

Comment:
Can't this be simplified to `attribute?.name.length > 0`?

Reply:
yes

Reply:
actually, we cannot. Updating, tho.

Reply:
https://github.com/woocommerce/woocommerce/pull/48928#discussion_r1659144279